### PR TITLE
Add testing coverage for pretty_print

### PR DESCRIPTION
**Pull request by gunnellEvan**
## Change Summary

Added test coverage to CoreSchema pretty print utility function

## Related issue number

#7656 

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by gunnellEvan** [comment]
please review

**Review by sydney-runkle — CHANGES_REQUESTED** [review]
@gunnellEvan,

Looks great, thanks for the contribution! I'm guessing some of the linting tests might fail, but you should be able to run `make format` and `make lint` locally to fix those issues!

**Comment by sydney-runkle** [comment]
Hmph, looks like there are still some errors (tests failing). Once that's resolved, all looks good!

File: tests/test_utils.py [hunk 1]
```diff
@@ -666,3 +666,66 @@ def walk(s, recurse):
...
+
+def test_pretty_print(capfd):
+    # Included metadata
+    schema = core_schema.AnySchema(
+        type='any',
+        ref='meta_schema',
+        metadata={'schema_type': 'any', 'test_id': '42'},
+        serialization=core_schema.simple_ser_schema('bool'),
+    )
+    expected_meta_info = "{\n    'type': 'any',\n    'ref': 'meta_schema',\n    'metadata': {'schema_type': 'any', 'test_id': '42'},\n    'serialization': {'type': 'bool'}\n}\n"
```
Comment (Viicos) [inline tests/test_utils.py:679]: Maybe multiline strings could be used?

Comment (sydney-runkle) [reply tests/test_utils.py:679]: @Viicos, yep, I think we're just waiting on formatting changes, but otherwise looks good!

Comment (gunnellEvan) [reply tests/test_utils.py:679]: Thanks for the suggestions! I'll go through and get the tests passing in the workflow later today and then re-request review.

Comment (gunnellEvan) [reply tests/test_utils.py]: Changes made and checks passed @sydney-runkle

**Review by sydney-runkle — APPROVED** [review]
Awesome work, we're glad to have a test for this!

**Pull request merged**